edge-modules/TwinTester/Program.cs

```diff
@@ -0,0 +1,344 @@
+            {
+                KeyValuePair<string, object> pair = (KeyValuePair<string, object>)twinUpdate;
+                eraseReportedProperties[pair.Key] = null; // erase twin property by assigning null
+            }
+
+            return eraseReportedProperties;
+        }
+
+        static bool IsPastFailureThreshold(DateTime twinUpdateTime)
+        {
+            return twinUpdateTime + Settings.Current.TwinUpdateFailureThreshold > DateTime.UtcNow;
+        }
+
+        static async Task CallAnalyzerToReportStatus(AnalyzerClient analyzerClient, string moduleId, string status, string responseJson)
+        {
+            try
+            {
+                await analyzerClient.AddTwinStatusAsync(new ResponseStatus { ModuleId = moduleId, StatusCode = status, ResultAsJson = responseJson, EnqueuedDateTime = DateTime.UtcNow });
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Failed call to report status to analyzer: {e}");
+            }
+        }
+
+        static async Task ValidateDesiredPropertyUpdates(RegistryManager registryManager, ModuleClient moduleClient, AnalyzerClient analyzerClient, Storage storage)
+        {
+            Twin receivedTwin;
+            try
+            {
+                receivedTwin = await moduleClient.GetTwinAsync();
+            }
+            catch (Exception e)
+            {
+                Logger.LogInformation($"Failed call to module client get twin: {e}");
+                return;
+            }
+
+            TwinCollection propertyUpdatesFromTwin = receivedTwin.Properties.Desired;
+            Dictionary<string, DateTime> desiredPropertiesUpdated = await storage.GetAllDesiredPropertiesUpdated();
+            Dictionary<string, DateTime> desiredPropertiesReceived = await storage.GetAllDesiredPropertiesReceived();
+            Dictionary<string, string> propertiesToRemoveFromTwin = new Dictionary<string, string>();
+            foreach (KeyValuePair<string, DateTime> desiredPropertyUpdate in desiredPropertiesUpdated)
+            {
+                bool doesTwinHaveUpdate = propertyUpdatesFromTwin.Contains(desiredPropertyUpdate.Key);
+                bool hasModuleReceivedCallback = desiredPropertiesReceived.ContainsKey(desiredPropertyUpdate.Key);
+                if (doesTwinHaveUpdate && hasModuleReceivedCallback)
+                {
+                    try
+                    {
+                        await storage.RemoveDesiredPropertyUpdate(desiredPropertyUpdate.Key);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.LogError($"Failed to remove validated reported property id {desiredPropertyUpdate.Key} from storage: {e}");
+                        continue;
+                    }
+
+                    string successStatus = $"{(int)StatusCode.Success}: Successfully sent/received desired property update";
```

Comment:
The direct method statuses reported to the analyzer have no messages associated with them. I chose to add a status message as well as the StatusCode's are manually defined.